cli/fathom_web/extract.py

```diff
@@ -0,0 +1,144 @@
+# These are MIME types the `mimetypes` library doesn't recognize or gets wrong. Matches were found at:
+# https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types
+MIME_TYPE_TO_FILE_EXTENSION = {
+    'application/font-woff': '.woff',
+    'application/font-woff2': '.woff2',
+    'application/vnd.ms-fontobject': '.eot',
+    'font/opentype': '.otf',
+    'font/ttf': '.ttf',
+    'font/woff2': '.woff2',
+    'image/jpeg': '.jpg',
+    'image/webp': '.webp',
+}
+
+
+@command()
+@option('--preserve-originals/--no-preserve-originals',
+        default=True,
+        help='Save original html files in a newly created `originals` directory in IN_DIRECTORY (default: True)')
+@argument('in_directory', type=Path(exists=True, file_okay=False))
+def main(in_directory, preserve_originals):
+    """
+    Extracts resources from the html pages in IN_DIRECTORY and stores them in a separate directory for Git-LFS storage.
+    """
+    if preserve_originals:
+        originals_dir = pathlib.Path(in_directory) / 'originals'
+        originals_dir.mkdir(parents=True, exist_ok=True)
+
+    for file in pathlib.Path(in_directory).iterdir():
+        if file.suffix != '.html':
+            print(f'Skipping {file.name}; not an html file')
+            continue
+        if preserve_originals:
+            shutil.copyfile(file, originals_dir / file.name)
+        extract_base64_data_from_html_page(file)
+
+
+def extract_base64_data_from_html_page(file: pathlib.Path):
+    """
+    Extract all base64 data from the given html page and store the data in separate files.
+
+    We do this by building a new html string using the non-base64 data pieces from the original file and the
+    filenames we will generate for each of the base64 data strings.
+
+    Base64 data is found with regex matching. Each data string is decoded and saved as a separate file.
+    """
+    with file.open(encoding='utf-8') as fp:
+        html = fp.read()
+
+    # Make the subresources directory
+    subresources_directory = file.parent / 'resources' / f'{file.stem}_resources'
+    subresources_directory.mkdir(parents=True, exist_ok=True)
+
+    offset = 0
+    new_html = ''
+
+    filename_counter = 0
+    # A cache for elements that are repeated (e.g. icons)
+    saved_strings = {}
+
+    # Remove any existing `<base>` tag
+    html = BASE_TAG_PATTERN.sub('', html)
+
+    # Add `'self'` to the Content Security Policy so we can load our extracted resources
+    html = OLD_CSP.sub(NEW_CSP, html)
+
+    base64_data_matches = BASE64_DATA_PATTERN.finditer(html)
+    for match in base64_data_matches:
+        # Add the content before the base64 data
+        new_html += html[offset:match.start(1)]
+
+        base64_string = match.group('string')
+
+        # Check to see if we have already encountered this base64 string
+        file_path = saved_strings.get(base64_string, None)
```

Comment:
`None` is the default, so we don't have to specify it.

Reply:
Wow, forgot about that. Thanks.